python/main.py

```diff
@@ -67,16 +75,67 @@ class AddItemResponse(BaseModel):
 
 # add_item is a handler to add a new item for POST /items .
 @app.post("/items", response_model=AddItemResponse)
-def add_item(
+async def add_item(
     name: str = Form(...),
     category: str = Form(...),
+    image: UploadFile = File(None),  # 画像を受け取る
     db: sqlite3.Connection = Depends(get_db),
 ):
     if not name:
         raise HTTPException(status_code=400, detail="name is required")
 
-    insert_item(Item(name=name, category=category))
-    return AddItemResponse(**{"message": f"item received: {name}"})
+    if not category:
+        raise HTTPException(status_code=400, detail="category is required")
+
+    if not images.exists(): # 画像
+        images.mkdir()
+
+    image_filename = "default.jpg"  # 画像がない場合はdefault.jpg
+
+
+    if image:
+        # 画像のデータを取得して SHA-256 でハッシュ化
+        image_data = await image.read()
+        hash_name = hashlib.sha256(image_data).hexdigest()
+        image_filename = f"{hash_name}.jpg"  # .jpg を付ける
```

Comment:
ただのコメント:
もしここを一行で書きたかったらhash_nameのところに関数を入れると一行でかけます！
```suggestion
        image_filename = f"{hashlib.sha256(image_data).hexdigest()}.jpg"
```